dbms/src/Storages/Page/V3/CheckpointFile/CPManifestFileReader.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/Page/V3/CheckpointFile/CPManifestFileReader.h>
+#include <Storages/Page/V3/CheckpointFile/ProtoHelper.h>
+
+#include <magic_enum.hpp>
+
+namespace DB::PS::V3
+{
+
+CheckpointProto::ManifestFilePrefix CPManifestFileReader::readPrefix()
+{
+    RUNTIME_CHECK_MSG(read_stage == ReadStage::ReadingPrefix, "unexpected read stage {}", magic_enum::enum_name(read_stage));
+
+    CheckpointProto::ManifestFilePrefix ret;
+    details::readMessageWithLength(*compressed_reader, ret);
+    read_stage = ReadStage::ReadingEdits;
+    return ret;
+}
+
+std::optional<universal::PageEntriesEdit> CPManifestFileReader::readEdits(CheckpointProto::StringsInternMap & strings_map)
```

Comment:
It is possible to use multiple threads to call this method in parallel in the future?

Reply:
Looks like we need to store the `offset` and `length` of each batch somewhere.

Reply:
May be better to consider split into multiple sub-manifest files, when one file is too large, if we want to increase parallelism.

Reply:
OK. I think we can do split job later.